vim-engine/src/main/kotlin/com/maddyhome/idea/vim/common/VimListenersNotifier.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2003-2023 The IdeaVim authors
+ *
+ * Use of this source code is governed by an MIT-style
+ * license that can be found in the LICENSE.txt file or at
+ * https://opensource.org/licenses/MIT.
+ */
+
+package com.maddyhome.idea.vim.common
+
+import com.maddyhome.idea.vim.api.VimEditor
+import com.maddyhome.idea.vim.state.mode.Mode
+
+public class VimListenersNotifier {
+  public val modeChangeListeners: MutableList<ModeChangeListener> = mutableListOf()
+  public val editorFocusListeners: MutableList<EditorFocusListener> = mutableListOf()
```

Comment:
These listeners should be migrated to kotlin flows at some moment, as they're more effective.